### PR TITLE
ISOM-1142: feat(modal): page create modal with validations

### PR DESCRIPTION
**Pull request by harishv7**
## Problem

We need to allow users to create new pages on Studio.

Closes ISOM-1142

## Solution

Implement Chakra modal with react-hook-form validations.

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [X] No - this PR is backwards compatible

## Before & After Screenshots

**AFTER**:
Screenshot 2024-06-05 at 1 55 14 PM

## Tests
- [ ] Ensure the fields display a dynamic counter of how chars are left
- [ ] Ensure validation for each field for min length of 1
- [ ] Ensure validation  max length of 100 chars for title
- [ ] Ensure validation of 150 chars for url

### CODE REVIEW

File: apps/studio/src/components/features/editing-experience/PageCreateModal.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<PageCreateFormFields>()
+
+  const watchAllFields = watch()
+  console.log(watchAllFields)
+  console.log(`errors`, errors)
```
Comment (seaerchin) [inline apps/studio/src/components/features/editing-experience/PageCreateModal.tsx:65]: could we remove this? i suspect you might have added it in for testing

File: apps/studio/src/components/features/editing-experience/PageCreateModal.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+  useEffect(() => {
+    setTitleLen(watchAllFields.title?.length || 0)
+    setPageUrlLen(watchAllFields.url?.length || 0)
+  }, [watchAllFields])
+
+  /* TODO: When integrating with BE */
+  const onSubmit: SubmitHandler<PageCreateFormFields> = (data) =>
+    console.log(data)
```
Comment (seaerchin) [inline apps/studio/src/components/features/editing-experience/PageCreateModal.tsx:73]: [FYI, no action required] the dto's PR on my end might have the trpc call ready -> just need to call it when merged

**Review by seaerchin — APPROVED** [review]
lgtm, no big issues on my end

File: apps/studio/src/components/features/editing-experience/PageCreateModal.tsx (hunk 1 above)
Comment (harishv7) [reply apps/studio/src/components/features/editing-experience/PageCreateModal.tsx]: thanks, removed!

File: apps/studio/src/components/features/editing-experience/PageCreateModal.tsx (hunk 2 above)
Comment (harishv7) [reply apps/studio/src/components/features/editing-experience/PageCreateModal.tsx]: gotcha

**Pull request merged**